gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/LoadBasedFlowEdgeImpl.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.modules.flow;
+
+import java.util.Properties;
+
+import org.jgrapht.graph.DefaultWeightedEdge;
+
+import org.apache.gobblin.util.ConfigUtils;
+import org.apache.gobblin.annotation.Alpha;
+import org.apache.gobblin.runtime.api.FlowEdge;
+import org.apache.gobblin.runtime.api.ServiceNode;
+import org.apache.gobblin.runtime.api.SpecExecutor;
+
+import lombok.Getter;
+
+/**
+ * A base implementation of a flowEdge in the weight multi-edge graph.
+ * For a weightedMultiGraph there could be multiple edges between two vertices.
+ * Recall that a triplet of <SourceNode, targetNode, specExecutor> determines one edge.
+ * It is expected that {@link org.jgrapht.graph.DirectedWeightedMultigraph#getAllEdges(Object, Object)}
+ * can return multiple edges with the same pair of source and destination but different SpecExecutor.
+ *
+ * Each edge has a {@FlowEdgeProp} which is used to evaluate a edge.
+ *
+ * The {@link LoadBasedFlowEdgeImpl} exposes additional interface to get/set load to edge directly.
+ * Load of edge is equivalent to weight defined in {@link DefaultWeightedEdge}.
+ * Since {@link #getWeight()} method is protected, {@link #getEdgeLoad()} will return the load.
+ * It is preferable to use {@link org.jgrapht.graph.DirectedWeightedMultigraph#setEdgeWeight(Object, double)}
```

Comment:
Not sure how weight logic is working here including its mutation. Can you please add more doc around it or references to jGraphT to make it clearer?

Reply:
Rewrote the javadoc.